File: statsig-java/java/src/main/java/com/statsig/ClientInitResponseOptions.java
package com.statsig;

import org.jetbrains.annotations.Nullable;

public class ClientInitResponseOptions {
  public HashAlgo hashAlgo;
  public String clientSDKKey;
  public boolean includeLocalOverrides;
  public GCIRResponseFormat responseFormat = GCIRResponseFormat.Initialize;
  private String hashAlgoInternal; // jni use string type
  private String responseFormatInternal; // jni use string type

  public ClientInitResponseOptions(
      @Nullable HashAlgo hashAlgo, String clientSDKKey, boolean includeLocalOverrides) {
    this.hashAlgo = hashAlgo;
    this.clientSDKKey = clientSDKKey;
    this.includeLocalOverrides = includeLocalOverrides;
    if (hashAlgo != null) {
      this.hashAlgoInternal = hashAlgo.convertToStr();
    }
  }

  public ClientInitResponseOptions(String clientSDKKey) {
    this(null, clientSDKKey, false);
  }

  public ClientInitResponseOptions(HashAlgo hashAlgo) {
    this(hashAlgo, null, false);
  }

  public ClientInitResponseOptions(HashAlgo hashAlgo, String clientSDKKey) {
    this(hashAlgo, clientSDKKey, false);
  }

  public ClientInitResponseOptions(HashAlgo hashAlgo, boolean includeLocalOverrides) {
    this(hashAlgo, null, includeLocalOverrides);
  }

  public ClientInitResponseOptions() {
    this(null, null, false);
  }

  public HashAlgo getHashAlgo() {
    return hashAlgo;
  }

  public void setHashAlgo(HashAlgo hashAlgo) {
    this.hashAlgo = hashAlgo;
    this.hashAlgoInternal = (hashAlgo != null) ? hashAlgo.convertToStr() : null;
  }

  public void setClientSDKKey(String clientSDKKey) {
    this.clientSDKKey = clientSDKKey;
  }

  public void setIncludeLocalOverrides(boolean includeLocalOverrides) {
    this.includeLocalOverrides = includeLocalOverrides;
  }

  public void setResponseFormat(GCIRResponseFormat responseFormat) {
    this.responseFormat = responseFormat;
    responseFormatInternal = responseFormat.convertToStr();
  }
}

File: statsig-go/statsig-cgo/src/client_init_response_options.go
package statsig

type ClientInitResponseOptions struct {
	HashAlgo              *HashAlgo           `json:"hash_algorithm,omitempty"`
	ClientSDKKey          *string             `json:"client_sdk_key,omitempty"`
	IncludeLocalOverrides *bool               `json:"include_local_overrides,omitempty"`
	ResponseFormat        *GCIRResponseFormat `json:"response_format,omitempty"`
	FeatureGateFilter     *[]string           `json:"feature_gate_filter,omitempty"`
	ExperimentFilter      *[]string           `json:"experiment_filter,omitempty"`
	DynamicConfigFilter   *[]string           `json:"dynamic_config_filter,omitempty"`
	LayerFilter           *[]string           `json:"layer_filter,omitempty"`
}
